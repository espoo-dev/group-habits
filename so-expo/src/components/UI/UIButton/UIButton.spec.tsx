import { render, screen } from '@testing-library/react-native';
import { IUIButtonProps, UIButton } from './index';

function setup(props: IUIButtonProps) {
  return render(<UIButton {...props} />)
}

describe('src > components > UI > UIButton', () => {
  it(`match snapshot`, () => {
    const tree = setup({ children: 'Button' }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly`, () => {
    setup({ children: 'Button' })

    const buttonText = screen.getByText(/button/i)
    const activityIndicator = screen.queryByTestId('spec-ui-button-activity-indicator')

    expect(buttonText).toBeOnTheScreen()
    expect(activityIndicator).toBeNull()
  })

  describe('when is loading', () => {
    it(`should render ActivityIndicator`, () => {
      setup({ children: 'Button', loading: true });

      const buttonText = screen.queryByText(/button/i)
      const activityIndicator = screen.getByTestId('spec-ui-button-activity-indicator')

      expect(buttonText).toBeNull()
      expect(activityIndicator).toBeOnTheScreen()
    });
  })
});
