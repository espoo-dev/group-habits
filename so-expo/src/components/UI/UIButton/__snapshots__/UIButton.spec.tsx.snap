// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > UI > UIButton match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      {
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      {
        "borderBottomColor": "transparent",
        "borderLeftColor": "transparent",
        "borderRightColor": "transparent",
        "borderTopColor": "transparent",
      },
      {
        "backgroundColor": "#4f46e5",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "display": "flex",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "textAlign": "center",
        },
        {
          "width": "100%",
        },
        {
          "fontWeight": "700",
        },
        {
          "color": "#fff",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;
